server/http_handler.go

```diff
@@ -70,26 +71,223 @@ type kvStore interface {
+	}
+
+	regionCache := tikvStore.GetRegionCache()
+
+	// init backOffer && infoSchema.
+	backOffer := tikv.NewBackoffer(context.Background(), 500)
+
+	return &httpHandlerTool{
+		regionCache: regionCache,
+		bo:          backOffer,
+		store:       tikvStore,
+	}
+}
+
+func (t *httpHandlerTool) getMvccByEncodedKey(encodedKey kv.Key) (*kvrpcpb.MvccGetByKeyResponse, error) {
+	keyLocation, err := t.regionCache.LocateKey(t.bo, encodedKey)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	tikvReq := &tikvrpc.Request{
+		Type: tikvrpc.CmdMvccGetByKey,
+		MvccGetByKey: &kvrpcpb.MvccGetByKeyRequest{
+			Key: encodedKey,
+		},
+	}
+	kvResp, err := t.store.SendReq(t.bo, tikvReq, keyLocation.Region, time.Minute)
+	log.Info(string(encodedKey), keyLocation.Region, string(keyLocation.StartKey), string(keyLocation.EndKey), kvResp, err)
+
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return kvResp.MvccGetByKey, nil
+}
+
+func (t *httpHandlerTool) getMvccByHandle(tableID, handle int64) (*kvrpcpb.MvccGetByKeyResponse, error) {
+	encodedKey := tablecodec.EncodeRowKeyWithHandle(tableID, handle)
+	return t.getMvccByEncodedKey(encodedKey)
+}
+
+func (t *httpHandlerTool) getMvccByStartTs(startTS uint64, startKey, endKey []byte) (*kvrpcpb.MvccGetByStartTsResponse, error) {
+	for {
+		curRegion, err := t.regionCache.LocateKey(t.bo, startKey)
+		if err != nil {
+			log.Error(startTS, startKey, err)
+			return nil, errors.Trace(err)
+		}
+
+		tikvReq := &tikvrpc.Request{
+			Type: tikvrpc.CmdMvccGetByStartTs,
+			MvccGetByStartTs: &kvrpcpb.MvccGetByStartTsRequest{
+				StartTs: startTS,
+			},
+		}
+		tikvReq.Context.Priority = kvrpcpb.CommandPri_Low
+		kvResp, err := t.store.SendReq(t.bo, tikvReq, curRegion.Region, time.Hour)
+		log.Info(startTS, string(startKey), curRegion.Region, string(curRegion.StartKey), string(curRegion.EndKey), kvResp)
+		if err != nil {
+			log.Error(startTS, string(startKey), curRegion.Region, string(curRegion.StartKey), string(curRegion.EndKey), err)
+			return nil, errors.Trace(err)
+		}
+		data := kvResp.MvccGetByStartTS
+		if err := data.GetRegionError(); err != nil {
+			log.Warn(startTS, string(startKey), curRegion.Region, string(curRegion.StartKey), string(curRegion.EndKey), err)
+			continue
+		}
+
+		if len(data.GetError()) > 0 {
+			log.Error(startTS, string(startKey), curRegion.Region, string(curRegion.StartKey), string(curRegion.EndKey), data.GetError())
+			return nil, errors.New(data.GetError())
+		}
+
+		key := data.GetKey()
+		if len(key) > 0 {
+			return data, nil
+		}
+
+		if len(endKey) > 0 && curRegion.Contains(endKey) {
+			return nil, nil
+		}
+		if len(curRegion.EndKey) == 0 {
+			return nil, nil
+		}
+		startKey = curRegion.EndKey
+	}
+}
+
+func (t *httpHandlerTool) getMvccByIdxValue(idx table.Index, values url.Values, idxCols []*model.ColumnInfo, handleStr string) (*kvrpcpb.MvccGetByKeyResponse, error) {
+	sc := new(stmtctx.StatementContext)
+	sc.TimeZone = time.UTC
```

Comment:
Maybe we should ask use to specify the timezone?
If the session use timezone A, here use timezone B, the decoded value does not match.